node/src/components/contract_runtime.rs

```diff
@@ -404,10 +408,40 @@ where
                 .ignore()
             }
             ContractRuntimeRequest::ExecuteBlock {
+                protocol_version,
+                execution_pre_state,
+                finalized_block,
+                deploys,
+                responder,
+            } => {
+                trace!(
+                    ?protocol_version,
+                    ?execution_pre_state,
+                    ?finalized_block,
+                    ?deploys,
+                    "execute block request"
+                );
+                let engine_state = Arc::clone(&self.engine_state);
+                let metrics = Arc::clone(&self.metrics);
+                async move {
+                    let result = operations::execute_finalized_block(
+                        engine_state.as_ref(),
+                        metrics.as_ref(),
+                        protocol_version,
+                        execution_pre_state,
+                        finalized_block,
+                        deploys,
+                    );
+                    trace!(?result, "execute block response");
+                    responder.respond(result).await
+                }
+                .ignore()
+            }
+            ContractRuntimeRequest::EnqueueBlockForExecution {
                 finalized_block,
                 deploys,
             } => {
-                info!(?finalized_block, "execute finalized block");
+                info!(?finalized_block, "enqueuing finalized block for execution");
                 if self.execution_pre_state.next_block_height == finalized_block.height() {
                     self.execute_finalized_block(
```

Comment:
Should we replace this call with `operations::execute_finalized_block` as well now?

Reply:
This call returns effects.  Ultimately, the plan is to move this functionality to `LinearChain`